bundler/lib/dependabot/bundler/native_helpers.rb

```diff
@@ -1,10 +1,47 @@
 # frozen_string_literal: true
 
+require "bundler"
+require "dependabot/shared_helpers"
+
 module Dependabot
   module Bundler
     module NativeHelpers
-      def self.helper_path
-        "bundle exec ruby #{File.join(native_helpers_root, 'run.rb')}"
+      def self.run_bundler_subprocess(function:, args:, bundler_version:)
+        # Run helper suprocess with all bundler-related ENV variables removed
+        with_unbundled_env do
+          SharedHelpers.run_helper_subprocess(
+            command: helper_path(bundler_version: bundler_version),
+            function: function,
+            args: args,
+            env: {
+              # Bundler will pick the matching installed major version
+              "BUNDLER_VERSION" => bundler_version,
+              "BUNDLE_GEMFILE" => File.join(versioned_helper_path(bundler_version: bundler_version), "Gemfile"),
+              "BUNDLE_PATH" => File.join(versioned_helper_path(bundler_version: bundler_version), ".bundle")
+            }
+          )
+        end
+      end
+
+      # TODO: Remove once Dependabot Core is shipped using Bundler >= v2.1.x
+      #
+      # This avoids variable behaviour versions in/out of containers
+      def self.with_unbundled_env(&block)
+        # > v2.1.x method
+        if ::Bundler.respond_to?(:with_unbundled_env)
```

Comment:
Flipped this around when v2 is installed it complained that `with_clean_env` was deprecated as it was still defined